interop-tests/src/test/scala/akka/grpc/scaladsl/RichErrorModelSpec.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2020-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc.scaladsl
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.grpc.internal.GrpcMetadataImpl
+import akka.grpc.{ GrpcClientSettings, Trailers }
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.stream.scaladsl.Source
+import akka.testkit.TestKit
+import com.typesafe.config.ConfigFactory
+import example.myapp.helloworld.grpc.helloworld._
+import io.grpc.StatusRuntimeException
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{ Duration, _ }
+import scala.concurrent.{ Await, ExecutionContext, Future }
+
+class RichErrorModelSpec
+    extends TestKit(ActorSystem("RichErrorSpec"))
+    with AnyWordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with ScalaFutures {
+
+  import com.google.protobuf.any.Any
+  import com.google.rpc.{ Code, Status }
+  import io.grpc.protobuf.StatusProto
+
+  implicit val sys: ActorSystem = system
+  implicit val ec: ExecutionContext = sys.dispatcher
+
+  object RichErrorImpl extends GreeterService {
+
+    // #rich_error_model_unary
+    private def toJavaProto(scalaPbSource: com.google.protobuf.any.Any): com.google.protobuf.Any = {
+      val javaPbOut = com.google.protobuf.Any.newBuilder
+      javaPbOut.setTypeUrl(scalaPbSource.typeUrl)
+      javaPbOut.setValue(scalaPbSource.value)
+      javaPbOut.build
+    }
+
+    def sayHello(in: HelloRequest): Future[HelloReply] = {
+      val status: Status = Status
+        .newBuilder()
+        .setCode(Code.INVALID_ARGUMENT.getNumber)
+        .setMessage("What is wrong?")
+        .addDetails(toJavaProto(Any.pack(new HelloReply("The password!"))))
```

Comment:
This should not be a `HelloReply` but one of the messages from https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto , right?

Reply:
It does not really matter what you put there, but seems ok to me to change.